python/lsst/ip/diffim/imageMapReduce.py

```diff
@@ -261,6 +261,7 @@ def run(self, mapperResults, exposure, **kwargs):
             patchMI = item.getMaskedImage()
             isNotNan = ~(np.isnan(patchMI.getImage().getArray()) |
                          np.isnan(patchMI.getVariance().getArray()))
```

Comment:
See my comment below about multiplying the arrays together to prevent needless loops. If you used np.isfinite instead of np.isnan you can save the inverting loop (if inf is also not a valid value).